File: src/main/java/corp/soyummy/shoppinglists/ShoppingListRepository.java
package corp.soyummy.shoppinglists;

import org.springframework.stereotype.Repository;
import org.springframework.data.mongodb.repository.MongoRepository;

import org.bson.types.ObjectId;
import java.util.List;

@Repository
public interface ShoppingListRepository extends MongoRepository<ShoppingList, String> {
    List<ShoppingList> findByOwner(ObjectId ownerId);
    List<ShoppingList> findByRecipeId(String recipeId);
    List<ShoppingList> findByStrIngredientContainingIgnoreCase(String strIngredient);
}

File: src/main/kotlin/com/soyummy/shoppinglists/ShoppingListRepository.kt
package com.soyummy.shoppinglists

import org.springframework.stereotype.Repository
import org.springframework.data.mongodb.repository.MongoRepository
import org.bson.types.ObjectId

@Repository
interface ShoppingListRepository : MongoRepository<ShoppingList, String> {
    fun findByOwner(ownerId: ObjectId): List<ShoppingList>
    fun findByRecipeIdAndOwner(recipeId: String, ownerId: String): List<ShoppingList>
    fun findByStrIngredientContainingIgnoreCaseAndOwner(strIngredient: String, ownerId: String): List<ShoppingList>
}